src/backend.rs

```diff
@@ -845,6 +903,13 @@ impl State {
         self.with_services(selectors, |service| {
             let service = service.borrow_mut();
             let mut tag_set = service.tags.borrow_mut();
+
+            if let Some(ref path) = self.db_path {
+                let store = TagStorage::new(&path);
```

Comment:
Opening the tags storage once per service sounds a bit overkill. Why not move the `TagStorage::new` to before `self.with_services`?